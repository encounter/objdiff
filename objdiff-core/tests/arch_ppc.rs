use objdiff_core::{
    diff::{self, display},
    obj,
    obj::SectionKind,
};

mod common;

#[test]
#[cfg(feature = "ppc")]
fn read_ppc() {
    let diff_config = diff::DiffObjConfig::default();
    let obj = obj::read::parse(include_object!("data/ppc/IObj.o"), &diff_config).unwrap();
    insta::assert_debug_snapshot!(obj);
    let symbol_idx =
        obj.symbols.iter().position(|s| s.name == "Type2Text__10SObjectTagFUi").unwrap();
    let diff = diff::code::no_diff_code(&obj, symbol_idx, &diff_config).unwrap();
    insta::assert_debug_snapshot!(diff.instruction_rows);
    let output = common::display_diff(&obj, &diff, symbol_idx, &diff_config);
    insta::assert_snapshot!(output);
}

#[test]
#[cfg(feature = "ppc")]
fn read_dwarf1_line_info() {
    let diff_config = diff::DiffObjConfig::default();
    let obj = obj::read::parse(include_object!("data/ppc/m_Do_hostIO.o"), &diff_config).unwrap();
    let line_infos = obj
        .sections
        .iter()
        .filter(|s| s.kind == SectionKind::Code)
        .map(|s| s.line_info.clone())
        .collect::<Vec<_>>();
    insta::assert_debug_snapshot!(line_infos);
}

#[test]
#[cfg(feature = "ppc")]
fn read_extab() {
    let diff_config = diff::DiffObjConfig::default();
    let obj = obj::read::parse(include_object!("data/ppc/NMWException.o"), &diff_config).unwrap();
    insta::assert_debug_snapshot!(obj);
}

#[test]
#[cfg(feature = "ppc")]
fn diff_ppc() {
    let diff_config = diff::DiffObjConfig::default();
    let mapping_config = diff::MappingConfig::default();
    let target_obj =
        obj::read::parse(include_object!("data/ppc/CDamageVulnerability_target.o"), &diff_config)
            .unwrap();
    let base_obj =
        obj::read::parse(include_object!("data/ppc/CDamageVulnerability_base.o"), &diff_config)
            .unwrap();
    let diff =
        diff::diff_objs(Some(&target_obj), Some(&base_obj), None, &diff_config, &mapping_config)
            .unwrap();

    let target_diff = diff.left.as_ref().unwrap();
    let base_diff = diff.right.as_ref().unwrap();
    let sections_display = display::display_sections(
        &target_obj,
        target_diff,
        display::SymbolFilter::None,
        false,
        false,
        true,
    );
    insta::assert_debug_snapshot!(sections_display);

    let target_symbol_idx = target_obj
        .symbols
        .iter()
        .position(|s| s.name == "WeaponHurts__20CDamageVulnerabilityCFRC11CWeaponModei")
        .unwrap();
    let target_symbol_diff = &target_diff.symbols[target_symbol_idx];
    let base_symbol_idx = base_obj
        .symbols
        .iter()
        .position(|s| s.name == "WeaponHurts__20CDamageVulnerabilityCFRC11CWeaponModei")
        .unwrap();
    let base_symbol_diff = &base_diff.symbols[base_symbol_idx];
    assert_eq!(target_symbol_diff.target_symbol, Some(base_symbol_idx));
    assert_eq!(base_symbol_diff.target_symbol, Some(target_symbol_idx));
    insta::assert_debug_snapshot!((target_symbol_diff, base_symbol_diff));
}

#[test]
#[cfg(feature = "ppc")]
fn read_vmx128_coff() {
    let diff_config = diff::DiffObjConfig { combine_data_sections: true, ..Default::default() };
    let obj = obj::read::parse(include_object!("data/ppc/vmx128.obj"), &diff_config).unwrap();
    insta::assert_debug_snapshot!(obj);
    let symbol_idx =
        obj.symbols.iter().position(|s| s.name == "?FloatingPointExample@@YAXXZ").unwrap();
    let diff = diff::code::no_diff_code(&obj, symbol_idx, &diff_config).unwrap();
    insta::assert_debug_snapshot!(diff.instruction_rows);
    let output = common::display_diff(&obj, &diff, symbol_idx, &diff_config);
    insta::assert_snapshot!(output);
}
