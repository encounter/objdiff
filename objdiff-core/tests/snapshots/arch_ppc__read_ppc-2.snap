---
source: objdiff-core/tests/arch_ppc.rs
expression: diff.instruction_rows
---
[
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 0,
                size: 4,
                opcode: 283,
                branch_dest: None,
            },
        ),
        kind: None,
        branch_from: None,
        branch_to: None,
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 4,
                size: 4,
                opcode: 260,
                branch_dest: None,
            },
        ),
        kind: None,
        branch_from: None,
        branch_to: None,
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 8,
                size: 4,
                opcode: 265,
                branch_dest: Some(
                    20,
                ),
            },
        ),
        kind: None,
        branch_from: None,
        branch_to: Some(
            InstructionBranchTo {
                ins_idx: 5,
                branch_idx: 0,
            },
        ),
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 12,
                size: 4,
                opcode: 263,
                branch_dest: None,
            },
        ),
        kind: None,
        branch_from: None,
        branch_to: None,
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 16,
                size: 4,
                opcode: 267,
                branch_dest: Some(
                    32,
                ),
            },
        ),
        kind: None,
        branch_from: None,
        branch_to: Some(
            InstructionBranchTo {
                ins_idx: 8,
                branch_idx: 1,
            },
        ),
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 20,
                size: 4,
                opcode: 264,
                branch_dest: None,
            },
        ),
        kind: None,
        branch_from: Some(
            InstructionBranchFrom {
                ins_idx: [
                    2,
                ],
                branch_idx: 0,
            },
        ),
        branch_to: None,
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 24,
                size: 4,
                opcode: 263,
                branch_dest: None,
            },
        ),
        kind: None,
        branch_from: None,
        branch_to: None,
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 28,
                size: 4,
                opcode: 323,
                branch_dest: None,
            },
        ),
        kind: None,
        branch_from: None,
        branch_to: None,
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 32,
                size: 4,
                opcode: 283,
                branch_dest: None,
            },
        ),
        kind: None,
        branch_from: Some(
            InstructionBranchFrom {
                ins_idx: [
                    4,
                ],
                branch_idx: 1,
            },
        ),
        branch_to: None,
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 36,
                size: 4,
                opcode: 445,
                branch_dest: None,
            },
        ),
        kind: None,
        branch_from: None,
        branch_to: None,
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 40,
                size: 4,
                opcode: 260,
                branch_dest: None,
            },
        ),
        kind: None,
        branch_from: None,
        branch_to: None,
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 44,
                size: 4,
                opcode: 265,
                branch_dest: Some(
                    56,
                ),
            },
        ),
        kind: None,
        branch_from: None,
        branch_to: Some(
            InstructionBranchTo {
                ins_idx: 14,
                branch_idx: 2,
            },
        ),
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 48,
                size: 4,
                opcode: 263,
                branch_dest: None,
            },
        ),
        kind: None,
        branch_from: None,
        branch_to: None,
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 52,
                size: 4,
                opcode: 267,
                branch_dest: Some(
                    68,
                ),
            },
        ),
        kind: None,
        branch_from: None,
        branch_to: Some(
            InstructionBranchTo {
                ins_idx: 17,
                branch_idx: 3,
            },
        ),
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 56,
                size: 4,
                opcode: 264,
                branch_dest: None,
            },
        ),
        kind: None,
        branch_from: Some(
            InstructionBranchFrom {
                ins_idx: [
                    11,
                ],
                branch_idx: 2,
            },
        ),
        branch_to: None,
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 60,
                size: 4,
                opcode: 263,
                branch_dest: None,
            },
        ),
        kind: None,
        branch_from: None,
        branch_to: None,
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 64,
                size: 4,
                opcode: 323,
                branch_dest: None,
            },
        ),
        kind: None,
        branch_from: None,
        branch_to: None,
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 68,
                size: 4,
                opcode: 283,
                branch_dest: None,
            },
        ),
        kind: None,
        branch_from: Some(
            InstructionBranchFrom {
                ins_idx: [
                    13,
                ],
                branch_idx: 3,
            },
        ),
        branch_to: None,
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 72,
                size: 4,
                opcode: 263,
                branch_dest: None,
            },
        ),
        kind: None,
        branch_from: None,
        branch_to: None,
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 76,
                size: 4,
                opcode: 260,
                branch_dest: None,
            },
        ),
        kind: None,
        branch_from: None,
        branch_to: None,
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 80,
                size: 4,
                opcode: 445,
                branch_dest: None,
            },
        ),
        kind: None,
        branch_from: None,
        branch_to: None,
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 84,
                size: 4,
                opcode: 265,
                branch_dest: Some(
                    96,
                ),
            },
        ),
        kind: None,
        branch_from: None,
        branch_to: Some(
            InstructionBranchTo {
                ins_idx: 24,
                branch_idx: 4,
            },
        ),
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 88,
                size: 4,
                opcode: 263,
                branch_dest: None,
            },
        ),
        kind: None,
        branch_from: None,
        branch_to: None,
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 92,
                size: 4,
                opcode: 267,
                branch_dest: Some(
                    108,
                ),
            },
        ),
        kind: None,
        branch_from: None,
        branch_to: Some(
            InstructionBranchTo {
                ins_idx: 27,
                branch_idx: 5,
            },
        ),
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 96,
                size: 4,
                opcode: 264,
                branch_dest: None,
            },
        ),
        kind: None,
        branch_from: Some(
            InstructionBranchFrom {
                ins_idx: [
                    21,
                ],
                branch_idx: 4,
            },
        ),
        branch_to: None,
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 100,
                size: 4,
                opcode: 263,
                branch_dest: None,
            },
        ),
        kind: None,
        branch_from: None,
        branch_to: None,
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 104,
                size: 4,
                opcode: 323,
                branch_dest: None,
            },
        ),
        kind: None,
        branch_from: None,
        branch_to: None,
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 108,
                size: 4,
                opcode: 283,
                branch_dest: None,
            },
        ),
        kind: None,
        branch_from: Some(
            InstructionBranchFrom {
                ins_idx: [
                    23,
                ],
                branch_idx: 5,
            },
        ),
        branch_to: None,
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 112,
                size: 4,
                opcode: 263,
                branch_dest: None,
            },
        ),
        kind: None,
        branch_from: None,
        branch_to: None,
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 116,
                size: 4,
                opcode: 260,
                branch_dest: None,
            },
        ),
        kind: None,
        branch_from: None,
        branch_to: None,
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 120,
                size: 4,
                opcode: 445,
                branch_dest: None,
            },
        ),
        kind: None,
        branch_from: None,
        branch_to: None,
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 124,
                size: 4,
                opcode: 265,
                branch_dest: Some(
                    136,
                ),
            },
        ),
        kind: None,
        branch_from: None,
        branch_to: Some(
            InstructionBranchTo {
                ins_idx: 34,
                branch_idx: 6,
            },
        ),
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 128,
                size: 4,
                opcode: 263,
                branch_dest: None,
            },
        ),
        kind: None,
        branch_from: None,
        branch_to: None,
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 132,
                size: 4,
                opcode: 267,
                branch_dest: Some(
                    148,
                ),
            },
        ),
        kind: None,
        branch_from: None,
        branch_to: Some(
            InstructionBranchTo {
                ins_idx: 37,
                branch_idx: 7,
            },
        ),
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 136,
                size: 4,
                opcode: 264,
                branch_dest: None,
            },
        ),
        kind: None,
        branch_from: Some(
            InstructionBranchFrom {
                ins_idx: [
                    31,
                ],
                branch_idx: 6,
            },
        ),
        branch_to: None,
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 140,
                size: 4,
                opcode: 263,
                branch_dest: None,
            },
        ),
        kind: None,
        branch_from: None,
        branch_to: None,
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 144,
                size: 4,
                opcode: 323,
                branch_dest: None,
            },
        ),
        kind: None,
        branch_from: None,
        branch_to: None,
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 148,
                size: 4,
                opcode: 263,
                branch_dest: None,
            },
        ),
        kind: None,
        branch_from: Some(
            InstructionBranchFrom {
                ins_idx: [
                    33,
                ],
                branch_idx: 7,
            },
        ),
        branch_to: None,
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 152,
                size: 4,
                opcode: 263,
                branch_dest: None,
            },
        ),
        kind: None,
        branch_from: None,
        branch_to: None,
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 156,
                size: 4,
                opcode: 445,
                branch_dest: None,
            },
        ),
        kind: None,
        branch_from: None,
        branch_to: None,
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 160,
                size: 4,
                opcode: 264,
                branch_dest: None,
            },
        ),
        kind: None,
        branch_from: None,
        branch_to: None,
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 164,
                size: 4,
                opcode: 263,
                branch_dest: None,
            },
        ),
        kind: None,
        branch_from: None,
        branch_to: None,
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 168,
                size: 4,
                opcode: 445,
                branch_dest: None,
            },
        ),
        kind: None,
        branch_from: None,
        branch_to: None,
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 172,
                size: 4,
                opcode: 263,
                branch_dest: None,
            },
        ),
        kind: None,
        branch_from: None,
        branch_to: None,
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 176,
                size: 4,
                opcode: 441,
                branch_dest: None,
            },
        ),
        kind: None,
        branch_from: None,
        branch_to: None,
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 180,
                size: 4,
                opcode: 323,
                branch_dest: None,
            },
        ),
        kind: None,
        branch_from: None,
        branch_to: None,
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 184,
                size: 4,
                opcode: 289,
                branch_dest: None,
            },
        ),
        kind: None,
        branch_from: None,
        branch_to: None,
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 188,
                size: 4,
                opcode: 265,
                branch_dest: Some(
                    196,
                ),
            },
        ),
        kind: None,
        branch_from: None,
        branch_to: Some(
            InstructionBranchTo {
                ins_idx: 49,
                branch_idx: 8,
            },
        ),
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 192,
                size: 4,
                opcode: 445,
                branch_dest: None,
            },
        ),
        kind: None,
        branch_from: None,
        branch_to: None,
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 196,
                size: 4,
                opcode: 442,
                branch_dest: None,
            },
        ),
        kind: None,
        branch_from: Some(
            InstructionBranchFrom {
                ins_idx: [
                    47,
                ],
                branch_idx: 8,
            },
        ),
        branch_to: None,
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 200,
                size: 4,
                opcode: 323,
                branch_dest: None,
            },
        ),
        kind: None,
        branch_from: None,
        branch_to: None,
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 204,
                size: 4,
                opcode: 289,
                branch_dest: None,
            },
        ),
        kind: None,
        branch_from: None,
        branch_to: None,
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 208,
                size: 4,
                opcode: 265,
                branch_dest: Some(
                    216,
                ),
            },
        ),
        kind: None,
        branch_from: None,
        branch_to: Some(
            InstructionBranchTo {
                ins_idx: 54,
                branch_idx: 9,
            },
        ),
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 212,
                size: 4,
                opcode: 445,
                branch_dest: None,
            },
        ),
        kind: None,
        branch_from: None,
        branch_to: None,
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 216,
                size: 4,
                opcode: 442,
                branch_dest: None,
            },
        ),
        kind: None,
        branch_from: Some(
            InstructionBranchFrom {
                ins_idx: [
                    52,
                ],
                branch_idx: 9,
            },
        ),
        branch_to: None,
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 220,
                size: 4,
                opcode: 323,
                branch_dest: None,
            },
        ),
        kind: None,
        branch_from: None,
        branch_to: None,
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 224,
                size: 4,
                opcode: 289,
                branch_dest: None,
            },
        ),
        kind: None,
        branch_from: None,
        branch_to: None,
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 228,
                size: 4,
                opcode: 265,
                branch_dest: Some(
                    236,
                ),
            },
        ),
        kind: None,
        branch_from: None,
        branch_to: Some(
            InstructionBranchTo {
                ins_idx: 59,
                branch_idx: 10,
            },
        ),
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 232,
                size: 4,
                opcode: 445,
                branch_dest: None,
            },
        ),
        kind: None,
        branch_from: None,
        branch_to: None,
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 236,
                size: 4,
                opcode: 442,
                branch_dest: None,
            },
        ),
        kind: None,
        branch_from: Some(
            InstructionBranchFrom {
                ins_idx: [
                    57,
                ],
                branch_idx: 10,
            },
        ),
        branch_to: None,
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 240,
                size: 4,
                opcode: 323,
                branch_dest: None,
            },
        ),
        kind: None,
        branch_from: None,
        branch_to: None,
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 244,
                size: 4,
                opcode: 289,
                branch_dest: None,
            },
        ),
        kind: None,
        branch_from: None,
        branch_to: None,
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 248,
                size: 4,
                opcode: 265,
                branch_dest: Some(
                    256,
                ),
            },
        ),
        kind: None,
        branch_from: None,
        branch_to: Some(
            InstructionBranchTo {
                ins_idx: 64,
                branch_idx: 11,
            },
        ),
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 252,
                size: 4,
                opcode: 445,
                branch_dest: None,
            },
        ),
        kind: None,
        branch_from: None,
        branch_to: None,
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 256,
                size: 4,
                opcode: 263,
                branch_dest: None,
            },
        ),
        kind: None,
        branch_from: Some(
            InstructionBranchFrom {
                ins_idx: [
                    62,
                ],
                branch_idx: 11,
            },
        ),
        branch_to: None,
        arg_diff: [],
    },
    InstructionDiffRow {
        ins_ref: Some(
            InstructionRef {
                address: 260,
                size: 4,
                opcode: 269,
                branch_dest: None,
            },
        ),
        kind: None,
        branch_from: None,
        branch_to: None,
        arg_diff: [],
    },
]
